src/back/spv/writer.rs

```diff
@@ -131,7 +131,7 @@ impl Writer {
             constant_ids: take(&mut self.constant_ids).recycle(),
             cached_constants: take(&mut self.cached_constants).recycle(),
             global_variables: take(&mut self.global_variables).recycle(),
-            cached: take(&mut self.cached).recycle(),
+            saved_cached: take(&mut self.saved_cached).recycle(),
```

Comment:
why doesn't `recycle()` do the `take` internally?

Reply:
The documentation for the `Recycle` trait explains the idea here: if the method takes a `&mut`, then that makes it easy for implementations to just update fields of interest, instead of constructing a fresh value so that we'll get compile-time errors if a field is forgotten. Passing by value encourages implementations to do the right thing.

Of course, none of this guarantees anything. An implementation could perfectly well just assign to a few fields of `self` and then return `self`. It's just a prompt. I couldn't figure out any way to enforce the rule that every field should be listed.

If you think it'd be okay for `Recyclable::recycle` to just take `&mut self` and remind people to do the `take` in the documentation, then we can do that.

Reply:
That's good reasoning, let's keep it